### PR TITLE
Replace x-forwarded support with Airlift version

### PR DESCRIPTION
**Pull request by dain**
Replace manual handling of forwarded headers with request rewriting in Airlift

### CODE REVIEW

**Comment by electrum** [comment]
Maybe say "Jetty" instead

File: presto-main/src/main/java/io/prestosql/server/security/SecurityConfig.java [hunk 1]
```diff
@@ -24,13 +24,15 @@
 import java.util.List;
 import java.util.Optional;
 
-@DefunctConfig("http.server.authentication.enabled")
+@DefunctConfig({
+        "http.server.authentication.enabled",
+        "http-server.authentication.allow-forwarded-https",
+        "dispatcher.forwarded-header"})
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/security/SecurityConfig.java:30]: Format with trailing comma
```java
        "dispatcher.forwarded-header",
})

File: presto-main/src/test/java/io/prestosql/server/ui/TestWebUi.java [hunk 2]
```diff
@@ -433,11 +434,11 @@ private static void testAlwaysAuthorized(URI baseUri, OkHttpClient authorizedCli
 
         assertOk(authorizedClient, getValidApiLocation(baseUri));
 
-        assertRedirect(authorizedClient, getLoginHtmlLocation(baseUri), getUiLocation(baseUri));
+        assertRedirect(authorizedClient, getLoginHtmlLocation(baseUri), getUiLocation(baseUri), false);
```
Comment (electrum) [inline presto-main/src/test/java/io/prestosql/server/ui/TestWebUi.java:437]: Why?

Comment (dain) [reply presto-main/src/test/java/io/prestosql/server/ui/TestWebUi.java:437]: The proxy test doesn't go over https so the protocol authentication does not work

**Pull request merged**